File: Python/TelemetrySystem/telemetry.py
from client import TelemetryClient

class TelemetryDiagnostics:
    DiagnosticChannelConnectionString = "*111#"

    def __init__(self):
        self._telemetry_client = TelemetryClient()
        self.diagnostic_info = ""

    def check_transmission(self):
        self.diagnostic_info = ""
        self._telemetry_client.disconnect()

        retryLeft = 3
        while ((not self._telemetry_client.online_status) and retryLeft > 0):
            self._telemetry_client.connect(TelemetryDiagnostics.DiagnosticChannelConnectionString)
            retryLeft -= 1

        if not self._telemetry_client.online_status:
            raise Exception("Unable to connect.")

        self._telemetry_client.send(TelemetryClient.DIAGNOSTIC_MESSAGE)
        self.diagnostic_info = self._telemetry_client.receive()

File: Ruby/TelemetrySystem/telemetry.rb
require './client'

class TelemetryDiagnostics
  attr_reader :diagnostic_info
  DIAGNOSTIC_CHANNEL_CONNECTION_STRING = "*111#"
  
  def initialize
    @telemetry_client = TelemetryClient.new
    @diagnostic_info = ""
  end

  def check_transmission
    @diagnostic_info = ""
    @telemetry_client.disconnect

    retry_left = 3
    while (not @telemetry_client.online_status) and retry_left > 0
      @telemetry_client.connect(DIAGNOSTIC_CHANNEL_CONNECTION_STRING)
      retry_left -= 1
    end

    if not @telemetry_client.online_status
      raise Exception.new("Unable to connect.")
    end

    @telemetry_client.send(TelemetryClient::DIAGNOSTIC_MESSAGE)
    @diagnostic_info = @telemetry_client.receive
  end
end
